Update recommendations for ScalarDB (#212)
scalar-labs/scalar-kubernetes: docs/CreateEKSClusterForScalarDBCluster.md

### Create at least three worker nodes and three pods

To ensure that the EKS cluster has high availability, you should use at least three worker nodes and deploy at least three pods spread across the worker nodes. You can see the [sample configurations](../conf/scalardb-cluster-custom-values-indirect-mode.yaml) of `podAntiAffinity` for making three pods spread across the worker nodes.

{% capture notice--info %}
**Note**

If you place the worker nodes in different [availability zones](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html) (AZs), you can withstand an AZ failure.
{% endcapture %}

<div class="notice--info">{{ notice--info | markdownify }}</div>

### Use 4vCPU / 8GB memory nodes for the worker node in the ScalarDB Cluster node group

From the perspective of commercial licenses, resources for one pod running ScalarDB Cluster are limited to 2vCPU / 4GB memory. In addition, some pods other than ScalarDB Cluster pods could exist on the worker nodes.

> In other words, the following components could run on one worker node:

Reviewer: Does the wording for this depend on how we word a similar sentence in a different PR?: [related comment](https://github.com/scalar-labs/scalar-kubernetes/pull/213/files#r1382880277)

* ScalarDB Cluster pod (2vCPU / 4GB)
* Envoy proxy (if you use `indirect` client mode or use a programming language other than Java)
* Your application pods (if you choose to run your application's pods on the same worker node)

> * Monitoring components (if you deploy monitoring components such `kube-prometheus-stack`)

Reviewer: ```suggestion
* Monitoring components (if you deploy monitoring components such as `kube-prometheus-stack`)
```

* Kubernetes components

{% capture notice--info %}
**Note**

You do not need to deploy an Envoy pod when using `direct-kubernetes` mode.
{% endcapture %}

<div class="notice--info">{{ notice--info | markdownify }}</div>

With this in mind, you should use a worker node that has at least 4vCPU / 8GB memory resources and use at least three worker nodes for availability, as mentioned in [Create at least three worker nodes and three pods](#create-at-least-three-worker-nodes-and-three-pods).

> However, three nodes with at least 4vCPU / 8GB memory resources per node is the minimum for a production environment. You should also consider the resources of the EKS cluster (for example, the number of worker nodes, vCPUs per node, memories per node, ScalarDB Cluster pods, and pods for your application), which depend on your system's workload. In addition, if you plan to scale the pods automatically by using some features like [Horizontal Pod Autoscaling (HPA)](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/), you should consider the maximum number of pods on the worker node to decide on the worker node resources.

Reviewer: ```suggestion
However, three nodes with at least 4vCPU / 8GB memory resources per node is the minimum for a production environment. You should also consider the resources of the EKS cluster (for example, the number of worker nodes, vCPUs per node, memories per node, ScalarDB Cluster pods, and pods for your application), which depend on your system's workload. In addition, if you plan to scale the pods automatically by using some features like [Horizontal Pod Autoscaling (HPA)](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/), you should consider the maximum number of pods on the worker node when deciding the worker node resources.
```

You should create the EKS cluster on a private network (private subnet in a VPC) since ScalarDB Cluster does not provide any services to users directly via internet access. We recommend accessing ScalarDB Cluster via a private network from your applications.

### Restrict connections by using some security features based on your requirements

You should restrict unused connections in ScalarDB Cluster. To restrict unused connections, you can use some security features in AWS, like [security groups](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html) and [network access control lists](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-network-acls.html).

> The connections (ports) that ScalarDB Cluster uses by default are as follows.

Reviewer: ```suggestion
The connections (ports) that ScalarDB Cluster uses by default are as follows:
```

    * 60053/TCP (load balancing for ScalarDB Cluster)
    * 9001/TCP (accepts monitoring requests for Scalar Envoy itself)

> {% capture notice--info %}
> **Note**
> 
> If you change the default listening port for ScalarDB Cluster in the configuration file (`scalardb-cluster-node.properties`), you must allow connections by using the port that you configured.
> {% endcapture %}
> 
> <div class="notice--info">{{ notice--info | markdownify }}</div>
> 
> {% capture notice--info %}
> **Note**
> 
> You also must allow the connections that EKS uses itself. For more details about Amazon EKS security group requirements, refer to [Amazon EKS security group requirements and considerations](https://docs.aws.amazon.com/eks/latest/userguide/sec-group-reqs.html).
> {% endcapture %}
> 
> <div class="notice--info">{{ notice--info | markdownify }}</div>

Reviewer: ```suggestion
{% capture notice--info %}
**Note**

- If you change the default listening port for ScalarDB Cluster in the configuration file (`scalardb-cluster-node.properties`), you must allow connections by using the port that you configured.
- You also must allow the connections that EKS uses itself. For more details about Amazon EKS security group requirements, refer to [Amazon EKS security group requirements and considerations](https://docs.aws.amazon.com/eks/latest/userguide/sec-group-reqs.html).
{% endcapture %}

<div class="notice--info">{{ notice--info | markdownify }}</div>
```
